### PR TITLE
[WIP] New File Structure + Version Picker

### PR DESCRIPTION
**Pull request by kognise**
This PR changes the file structure to `water.css`, `dark.css`, and `light.css`. It also introduces a full rewrite of the version picker on the documentation site.

Todo:
- [x] Add fancy animations, @jonaskuske :)
- [x] Wire up the copy to clipboard button
- [x] Update the readme with the new files

Are there any other changes in scope of this pull request?

### CODE REVIEW

**Comment by kognise** [comment]
What happened to Netlify? Huh.

**Comment by kognise** [comment]
Looks like Netlify isn't deploying this because it's a PR to the `two` branch, not `master`.

**Comment by jonaskuske** [comment]
Are those todos up for grabs? I could work on them tomorrow @kognise :)

**Comment by kognise** [comment]
> Are those todos up for grabs?

Absolutely! I just [added you as a collaborator](https://github.com/kognise/water.css/invitations) so you can push directly to this branch.

**Comment by jonaskuske** [comment]
@kognise Awesome, thank you!

File: README.md [hunk 1]
```diff
@@ -54,52 +59,54 @@ Just stick this in your `<head>`:
...
+For the main `water.css` file, dark is only treated as a _default theme_: if a user has a preference for either dark or light mode on their device, the stylesheet will respect this. This detection is made possible through a recent CSS media query called `prefers-color-scheme`. If you want to avoid this behavior and enforce dark or light theme, use either `dark.css` or `light.css`.
 
 #### Want to support Internet Explorer?
 
-Sure, just extend the theme prefix with `-legacy`, e.g. `dark-legacy.min.css`.
-Be aware that these versions **do not support** [runtime theming](#theming) as they use hard coded values rather than variables. Additionally, if you use a legacy version that is not standalone, we recommend [you add the respective preload tags to improve load times](https://watercss.netlify.com/?legacy#installation).
+Both `dark.css` and `light.css` support Internet Explorer 11, but the main `water.css` file **does not** due to lack of `prefers-color-scheme` support.
```
Comment (jonaskuske) [inline README.md:66]: I think we should mention here that the main water.css *does* support IE, it just won't switch between dark and light automatically. (right?)
With the current wording it sounds like choosing water.css means that users on IE will see broken styles.

Comment (kognise) [reply README.md:66]: Oh yeah, good point. I'll fix that, thanks!

Comment (kognise) [reply README.md]: @jonaskuske I updated that section to the following, does it make more sense?

All three distributions of Water.css support Internet Explorer 11, but the main `water.css` file **doesn't respect the user's color scheme** and will be locked to dark mode due to lack of `prefers-color-scheme` support.

Comment (jonaskuske) [reply README.md]: LGTM 👍

**Comment by kylejrp** [comment]
> Looks like Netlify isn't deploying this because it's a PR to the `two` branch, not `master`.

It looks like there's ways to configure Netlify to allow for building of other branches. See https://docs.netlify.com/site-deploys/overview/#branch-deploy-controls

**Comment by jonaskuske** [comment]
Hmm, are we not using prettier or is it just not part of this branch?

My VSCode completely wrecks havoc on the code formatting when saving 😅
If we're not using it, I'll add a `.prettierignore` with `*`?

**Comment by kognise** [comment]
Yeah, we aren't using it - a prettierignore would be cool. I uninstalled the extension a while ago because it got really annoying for non-prettier projects.

File: package.json [hunk 2]
```diff
@@ -70,24 +70,10 @@
   "browserslist": {
     "legacy": [
       "defaults AND not android 4.4.3"
-    ],
-    "modern": [
```
[the remark this answers is not in the document]
Comment (kognise) [reply package.json:74]: Aren't we still doing the legacy prefixing?

**Comment by jonaskuske** [comment]
Added, I'll push my changes soon 👍🏻

Do we still need the "Browser support" row in the info table? All versions support Internet Explorer now.

**Comment by kognise** [comment]
Water.css doesn't support prefers-color-scheme in IE, I thought we might want to include that. Thanks!

**Comment by jonaskuske** [comment]
Hmm, same story as with the Readme though: if we say it's only compatible with current browsers, that sounds like the styling doesn't work in IE although it does. 😅

I think we can just remove that row from the info table tbh. We mention that the "Auto" feature depends on `prefers-media-query`, that should be enough to connect the dots. I've also made `prefers-media-query` clickable, it now links to the MDN page. (which also has a browser support table)

**Comment by jonaskuske** [comment]
@kognise I went ahead and removed the table row, but updated the "Theme" info to:

> Respects user-defined theme settings using `prefers-color-scheme`.
> Light in browsers where the theme settings can't be detected.

If you disagree with these changes we can just revert the last commit :)

**Comment by kognise** [comment]
Do you still want the cool transitions? I'm not sure the best way to add them. Otherwise, :shipit:

**Comment by jonaskuske** [comment]
Which ones do you mean? The transition for updating the code snippet is there (...isn't is? Works in all browsers I tested including IE) and I thought transitioning the clipboard confirmation in and out wasn't worth the effort 😅

**Comment by kognise** [comment]
My b! It's there 😛 

Although it looks like it clips outside the container when it's enlarged - is there a simple way to fix that and is it worth fixing?

**Comment by jonaskuske** [comment]
We could prevent it from growing larger than scale 1 by adjusting the cubic bezier curve that controls the timing. (probably won't look as bouncy though then)
They're a bit like dark magic to me, but I can give it a try 😄

**Comment by jonaskuske** [comment]
Done! Not bouncing anymore, thus also not clipping anymore. Fine imo.

**Comment by kognise** [comment]
Awesome, I was just about to push the same change 😂 

For some reason GitHub won't let me review this, would you mind approving it?

**Comment by jonaskuske** [comment]
Noticed another bug, 1 sec 😅

**Comment by kognise** [comment]
Oof, missed that.

**Pull request merged**